server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/globalstate/SnapshotGlobalStateRequest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.cluster.snapshots.globalstate;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.support.master.MasterNodeRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.snapshots.SnapshotId;
+import org.elasticsearch.tasks.CancellableTask;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.tasks.TaskId;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+/**
+ * Get snapshot global state status request
+ */
+public class SnapshotGlobalStateRequest extends MasterNodeRequest<SnapshotGlobalStateRequest> {
+
+    private String repository = "_all";
+
+    private String snapshot;
+
+    public SnapshotGlobalStateRequest() {}
+
+    /**
+     * Constructs a new get snapshots request with given repository name and list of snapshots
+     *
+     * @param repository repository name
+     * @param snapshot  snapshot name
+     */
+    public SnapshotGlobalStateRequest(String repository, String snapshot) {
+        this.repository = repository;
+        this.snapshot = snapshot;
+    }
+
+    public SnapshotGlobalStateRequest(StreamInput in) throws IOException {
+        super(in);
+        repository = in.readString();
+        snapshot = in.readString();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(repository);
+        out.writeString(snapshot);
+    }
```

Comment:
Since this is a new action, we need to avoid sending it to older nodes because they won't understand it. That means we need to introduce a new constant in `TransportVersions` and then check the remote node version using `out.getTransportVersion().onOrAfter(...)`. If it's not new enough, throw an `IllegalArgumentException`.

Reply:
Right now I've put a check on `SnapshotGlobalStateResponse`. Am I missing something if I don't add transport version validation in `SnapshotGlobalStateRequest`?

Reply:
Once the above changes are fine-tuned, I'll make corresponding changes for test cases and docs.